### ISSUE TITLE
Tilemap, changing the depth of all tiles in a layer make some tiles disappear

### ISSUE THREAD

**Issue by Blist** [underscorediscovery/luxe#364]
Hi,

I have a map with multiple layer. I want to make the depth of a layer behind my player, but I have a bug that occurs randomly. When I want to change the depth of a whole layer some tiles disappear. 

Important thing: I don't have this bug on cpp build just with web build.

my code to update the depth
```haxe
public function uplayer(name:String, depth:Float) {
        var layer:luxe.tilemaps.TilemapVisualLayerGeometry = tilemap.visual.geometry_for_layer(name);
        for (row in layer) {
            for (tile in row) {
                if (tile != null) {
                    tile.depth = depth;
                }
            }
        }
    }
```

A snippet to reproduce the bug : 
```haxe
import luxe.Input;
import phoenix.Texture;
import luxe.tilemaps.Tilemap;

class Main extends luxe.Game {
    var tilemap:Tilemap;

    override function ready() {
        tilemap = new Tilemap({
            x: 0,
            y: 0,
            w: 16,
            h: 16,
            tile_width: 16,
            tile_height: 16,
            orientation: TilemapOrientation.ortho,
        });

        Luxe.resources.texture("assets/tiles.png").filter_mag = FilterType.nearest;

        tilemap.add_tileset({
            tile_width: 16,
            tile_height: 16,
            texture: Luxe.resources.texture("assets/tiles.png"),
            name: "default",
            first_id: 0,
        });

        tilemap.add_layer({
            name: "one",
            visible: true,
            opacity: 1,
            layer: 1,
        });

        tilemap.add_tiles_fill_by_id("one",1);

        tilemap.display({
            filter: FilterType.nearest,
            scale: 3,
        });

        uplayer("one", 3);

    }

    override function config(config:luxe.AppConfig) {
        config.preload.textures.push({id:"assets/tiles.png"});
        return config;
    }

    public function uplayer(name:String, depth:Float) {
        var layer:luxe.tilemaps.TilemapVisualLayerGeometry = tilemap.visual.geometry_for_layer(name);
        for (row in layer) {
            for (tile in row) {
                if (tile != null) {
                    tile.depth = depth;
                }
            }
        }
    }
}
```

the tile is use for the snippet (from PentaHelix's issue)

![tiles.png](https://camo.githubusercontent.com/d549a02fe861eb0728fba8001a3535a92a78e9af/687474703a2f2f692e696d6775722e636f6d2f6b4776417357592e706e67)

**Comment by Blist** [comment]
Some strange thing,

if I trace the transform.pos of each geometry that I have update, holes disappear ...

Without trace :
![without](http://imgur.com/ovXaapH.png)
With trace :
![with](http://imgur.com/ookkECX.png)

Snippet

```haxe 
public function uplayer(name:String, depth:Float) {
        var layer:luxe.tilemaps.TilemapVisualLayerGeometry = tilemap.visual.geometry_for_layer(name);
        for (row in layer) {
            for (tile in row) {
                if (tile != null) {
                    tile.depth = depth;
                    trace(tile.transform.pos);
                }
            }
        }
    }
```

Maybe, tilemap refresh his display before that all tiles are updates ?

**Comment by KeyMaster-** [comment]
Fixed with 5c38950 and 596c821. Geometry sorting was getting confused due to some subtle bugs.
